sdk/core/core/inc/_az_token_credential.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_TOKEN_CREDENTIAL_H
+#define AZ_TOKEN_CREDENTIAL_H
+
+#include <az_credential.h>
+#include <az_mut_span.h>
+#include <az_span.h>
+#include <az_span_builder.h>
+
+#include <time.h>
+
+#include <_az_cfg_prefix.h>
+
+enum {
+  _az_TOKEN_CREDENTIAL_TOKEN_BUFFER_SIZE = 3 * (1024 / 2),
+};
+
+typedef struct {
+  az_credential credential;
+  az_span _resource;
```

Comment:
We should use AADv2 scopes rather than AADv1 resource strings.

Reply:
Will do.